node/impl/eth/utils.go

```diff
@@ -703,53 +646,65 @@ func encodeAsABIHelper(param1 uint64, param2 uint64, data []byte) []byte {
 	return buf
 }
 
-func getTipsetByBlockNumber(ctx context.Context, cs ChainStore, blkParam string, strict bool) (*types.TipSet, error) {
-	if blkParam == "earliest" {
-		return nil, xerrors.New("block param \"earliest\" is not supported")
-	}
-
-	head := cs.GetHeaviestTipSet()
-	switch blkParam {
-	case "pending":
-		return head, nil
-	case "latest":
-		parent, err := cs.GetTipSetFromKey(ctx, head.Parents())
-		if err != nil {
-			return nil, xerrors.New("cannot get parent tipset")
-		}
-		return parent, nil
-	case "safe":
-		latestHeight := head.Height() - 1
-		safeHeight := latestHeight - ethtypes.SafeEpochDelay
-		ts, err := cs.GetTipsetByHeight(ctx, safeHeight, head, true)
+// decodePayload is a utility function which decodes the payload using the given codec
```

Comment:
this and `decodeParams` were just moved into utils, they're used by more than one module here